### PR TITLE
Issue #4250: Fix new TeamCity violations (part1)

### PR DESCRIPTION
**Pull request by romani**
#4250

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java [hunk 1]
```diff
@@ -74,15 +74,15 @@ public void testMixedCaseSpaces() {
 
     @Test
     public void testMixedCaseSpacesWithDifferentLocales() {
-        final Locale[] differentLocales = {new Locale("TR", "tr") };
-        final Locale defaultLocale = Locale.getDefault();
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java]: @romani We went through this before.
This is breaking the test as we have to save `defaultLocale` before we modify it. By the time `finally` is called, it is already modified.

File: src/test/java/com/puppycrawl/tools/checkstyle/api/SeverityLevelTest.java [hunk 2]
```diff
@@ -71,15 +71,15 @@ public void testMixedCaseSpaces() {
 
     @Test
     public void testMixedCaseSpacesWithDifferentLocales() {
-        final Locale[] differentLocales = {new Locale("TR", "tr") };
-        final Locale defaultLocale = Locale.getDefault();
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/SeverityLevelTest.java]: same here.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java [hunk 3]
```diff
@@ -87,11 +87,11 @@
      * @return the default alias for the given check
      */
     public static String getDefaultAlias(String sourceName) {
-        final int startIndex = sourceName.lastIndexOf('.') + 1;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java]: if we want `final` variables closer to where we are used, should we turn this on for our own check?
http://checkstyle.sourceforge.net/config_coding.html#VariableDeclarationUsageDistance
`ignoreFinal` property.

File: config/intellij-idea-inspections.xml [hunk 4]
```diff
@@ -501,7 +504,7 @@
     <inspection_tool class="Dependency" enabled="true" level="WARNING" enabled_by_default="true" />
     <inspection_tool class="DeprecatedCallableAddReplaceWith" enabled="true" level="ERROR" enabled_by_default="true" />
     <inspection_tool class="DeprecatedClassUsageInspection" enabled="false" level="WARNING" enabled_by_default="false" />
-    <inspection_tool class="DeprecatedIsStillUsed" enabled="true" level="WARNING" enabled_by_default="true" />
+    <inspection_tool class="DeprecatedIsStillUsed" enabled="true" level="WARNING" enabled_by_default="false" />
```
Comment (rnveach) [inline config/intellij-idea-inspections.xml]: what is the purpose of changing `enabled_by_default`?
If you are trying to disable it, shouldn't `enabled` be changed to `false`?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java (hunk 3 above)
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java]: we can not , Check is not ready for this.

File: config/intellij-idea-inspections.xml (hunk 4 above)
Comment (romani) [inline config/intellij-idea-inspections.xml]: I do not know, we will think about it later

File: src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java (hunk 1 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/ScopeTest.java]: reverted at #4286

File: src/test/java/com/puppycrawl/tools/checkstyle/api/SeverityLevelTest.java (hunk 2 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/api/SeverityLevelTest.java]: reverted at #4286

**Pull request closed without merging**